tests/main/snapshot-basic/task.yaml

```diff
@@ -125,3 +125,14 @@ execute: |
     # the io.Copy uses a 32k buffer, so extra memory usage should be limited.
     # The threshold in this test is set to about 40MB
     test "$(cat memory-kb.txt)" -lt 40000
+
+    # check that snapshot set id from the filename has authority
+    snap install test-snapd-sh
+    snap save test-snapd-sh
+    SNAPSHOT_FILE=$(ls /var/lib/snapd/snapshots/*test-snapd-sh*.zip)
+    NEWSNAPSHOT_FILE=$(echo "$SNAPSHOT_FILE" | sed -e's/[1-9]\+_/123_/')
+    # rename the snapshot file to force a new set id.
+    mv "$SNAPSHOT_FILE" "$NEWSNAPSHOT_FILE"
+    snap saved | MATCH "123 +test-snapd-sh"
```

Comment:
Should we also verify that a snapshot is only listed once with the new set ID?

Reply:
Added.